sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncDatabase.java

```diff
@@ -627,7 +604,7 @@ public CosmosAsyncContainer getContainer(String id) {
      * obtained users or an error.
      */
     public CosmosPagedFlux<CosmosUserProperties> queryUsers(String query, CosmosQueryRequestOptions options) {
-        return queryUsers(new SqlQuerySpec(query), options);
+        return queryUsersInternal(new SqlQuerySpec(query), options);
```

Comment:
There are two apis  for queryUsers, with option and without . With option, it need to be non null. Existing design

Reply:
This was discussed in API review versions earlier, we want these overriding APIs to provide better public surface access to end user. 
If the user provides `null` with this API, its their fault, because the overriding API exists which doesn't take `CosmosQueryRequestOptions`

Reply:
[Not Blocking for this PR]
Interesting..So by "user fault" you mean its acceptable for Cosmos API's to throw an NPE?

Just curious why don't we do a null check upfront and let it flow through the code.  If it is an expected param, you could use a  `Objects.requireNonNull(..)`

Reply:
A method that returns a reactor publisher (CosmosPagedFlux here) should not throw an exception directly. The error has to flow through the error channel and the user should be able to use `onError` handler to gracefully handle exceptions. Also, since `options` are optional, the user should still be able to pass a `null` value and this method should not throw.

Reply:
tracking in https://github.com/Azure/azure-sdk-for-java/issues/13298